feature: add readme for preview-build and fingerprint action (#232)
expo/expo-github-action: preview-build/README.md

<div align="center">
  <h1>preview-build</h1>

>   <p>Create [EAS Builds](https://docs.expo.dev/build/introduction/) for pull requests using [`@expo/fingerprint`](https://www.npmjs.com/package/@expo/fingerprint)</p>

Reviewer: <img width="990" alt="image" src="https://github.com/expo/expo-github-action/assets/1203991/44ec78fe-bd0b-4a8a-863e-98b744273f49">

```suggestion
  <p>Checking project fingerprinting for pull requests using <a href="https://www.npmjs.com/package/@expo/fingerprint"><code>@expo/fingerprint</code></a></p>
```

## Usage

To use this action, add the following code to your workflow:

```yaml
on:
  push:
    # REQUIRED: push main(default) branch is necessary for this action to update its fingerprint database
    branches: [main]
  pull_request:
    types: [opened, synchronize]

jobs:
  <JOB_NAME>:
    runs-on: <RUNNER>
    # REQUIRED: limit concurrency when pushing main(default) branch to prevent conflict for this action to update its fingerprint database
    concurrency: fingerprint-${{ github.event_name != 'pull_request' && 'main' || github.run_id }}
    permissions:
      # REQUIRED: Allow comments of PRs
      pull-requests: write # Allow comments on PRs
      # REQUIRED: Allow updating fingerprint in acton caches
      actions: write

    steps:
      - name: Setup EAS
        uses: expo/expo-github-action@v8
        with:
          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Create preview builds if fingerprint changed

>         uses: expo/expo-github-action/preview-build

Reviewer: ```suggestion
        uses: expo/expo-github-action/preview-build@main
```

Here is a summary of all the input options you can use.

| variable                           | default                                        | description                                                                                    |
| ---------------------------------- | ---------------------------------------------- | ---------------------------------------------------------------------------------------------- |
| **command**                        | -                                              | EAS CLI command to run when creating builds                                                    |
| **comment**                        | `true`                                         | If the action should summarize the EAS Update information as comment on a pull request         |
| **comment-id**                     | _[see code][code-defaults]_                    | unique id template to prevent duplicate comments ([read more](#preventing-duplicate-comments)) |
| **working-directory**              | -                                              | The relative directory of your Expo app                                                        |

> | **packager**                       | - `yarn`                                       | The package manager used to install the fingerprint tools                                      |

Reviewer: ```suggestion
| **packager**                       | `yarn`                                       | The package manager used to install the fingerprint tools                                      |
```

| **iosAppVersion**     | Version of the iOS app                                                                            |

## Example workflows

Here's an example workflow that uses this action to comment pull requests and create EAS Builds if fingerprint changed

```yaml
name: Build preview for pull requests

on:
  push:
    # REQUIRED: push main(default) branch is necessary for this action to update its fingerprint database
    branches: [main]
  pull_request:
    types: [opened, synchronize]

jobs:
  build:
    runs-on: ubuntu-latest
    # REQUIRED: limit concurrency when pushing main(default) branch to prevent conflict for this action to update its fingerprint database
    concurrency: fingerprint-${{ github.event_name != 'pull_request' && 'main' || github.run_id }}
    permissions:
      # REQUIRED: Allow comments of PRs
      pull-requests: write # Allow comments on PRs
      # REQUIRED: Allow updating fingerprint in acton caches
      actions: write
    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: 🏗  Setup EAS

>         uses: expo/expo-github-action

Reviewer: ```suggestion
        uses: expo/expo-github-action@v8
```

        with:
          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Install dependencies
        run: yarn install

      - name: Create preview builds if needed

>         uses: expo/expo-github-action/preview-build

Reviewer: ```suggestion
        uses: expo/expo-github-action/preview-build@main
```

        with:
          command: eas build --profile development --platform all
```

This workflow listens for pull request events and generates a fingerprint for each pull request. It then uses this action to create an EAS Build for the pull request, and deploys the build using another action.

## Caveats

### Preventing duplicate comments

When automating these preview comments, you have to be careful not to spam a pull request on every successful run.
Every comment contains a generated **message-id** to identify previously made comments and update them instead of creating a new comment.

### GitHub tokens

When using the GitHub API, you always need to be authenticated.
This action tries to auto-authenticate using the [Automatic token authentication][link-gha-token] from GitHub.

> You can overwrite the token by adding the `GITHUB_TOKEN` environment variable or add the **github-token** input.

Reviewer: ```suggestion
You can overwrite the token by adding the `GITHUB_TOKEN` environment variable or add the **github-token** input.

<div align="center">
  <br />
  with :heart:&nbsp;<strong>Expo</strong>
  <br />
</div>
```
